### DISCUSSION TITLE
O_DIRECT with XFS and fallocate - what are the guarantees?

### DISCUSSION THREAD

**Discussion by romange** [axboe/liburing#1114] [Q&A]
Maybe slightly off-topic but the team behind the project knows the answer for sure.
What are guarantees of the following sequence on XFS with NVME SSD?

1. fallocate RANGE -> wait for completion
2. WRITE(DIRECT_IO, subrange)  - wait for completion

Specifically, can we assume that the written data arrived to SSD device (possibly its cache memory), therefore the data can survive a software crash from this point onwards?

**Comment by axboe** [comment]
This isn't just slightly off-topic, it's very off-topic as it really has no relation to liburing or io_uring at all.

**Reply by romange** [reply]
you are right, closing
